testfixtures/comparison.py

```diff
@@ -685,3 +685,27 @@ def diff(x, y, x_label='', y_label=''):
             y_label or 'second',
             lineterm='')
     )
+
+
+class RangeComparison:
+    """
+    An object that can be used in comparisons of numerics to validate
+    actual falls within a certain range
+    """
+    def __init__(self, lower_bound, upper_bound):
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+
+    def __eq__(self, other):
+        try:
+            0 + other
+        except TypeError:
+            raise TypeError('Cannot detect range of %r' % (type(other)))
+        return self.lower_bound <= other <= self.upper_bound
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self):
+        return '<less than %s or greater than %s>' % (self.lower_bound,
```

Comment:
I'd prefer a repr along the lines of `<Range: [%s, %s]>` as it's more succinct and matches the [Postgres range notation](http://www.postgresql.org/docs/9.2/static/rangetypes.html#RANGETYPES-IO). That also gives us an obvious way to specify inclusive or exclusive boundaries, which would be a nice to have...